### PR TITLE
Automatically add a PR comment with documentation link

### PR DESCRIPTION
**Pull request by keith-zephyr**
When the HTML documentation is rebuilt, the link to generated documents is hard to find.  Automatically add a comment to the PR with the docs link.

### CODE REVIEW

File: .github/workflows/doc-build.yml [hunk 1]
```diff
@@ -23,6 +23,9 @@ on:
     - '.github/workflows/doc-build.yml'
     - 'scripts/dts/**'
     - 'scripts/requirements-doc.txt'
+  pull_request_target:
+    paths:
+    - 'doc/**'
```
Comment (gmarull) [inline .github/workflows/doc-build.yml:28]: `pull_request_target` is a source of security problems. We should not be using it unless it is strictly needed.

**Review by gmarull — CHANGES_REQUESTED** [review]
We already do https://github.com/zephyrproject-rtos/zephyr/blob/main/.github/workflows/doc-build.yml#L107

which results in:
https://github.com/zephyrproject-rtos/zephyr/actions/runs/3756326030/attempts/1#summary-10249646514

**Review by stephanosio — CHANGES_REQUESTED** [review]
I am not a big fan of every workflow adding their own comment -- it creates unnecessary noise and clutters the comment section.

This should not be necessary since we already provide a link to the built documentation in the form of annotation.

**Comment by keith-zephyr** [comment]
> I am not a big fan of every workflow adding their own comment -- it creates unnecessary noise and clutters the comment section.
> 
> This should not be necessary since we already provide a link to the built documentation in the form of annotation.

The annotation isn't easy to discover from the PR.  You have to find the Documentation Build in the list of checks, click "details", then click summary, and finally open the doc link.
 
I was hoping to make the doc link more visible on PRs that update docs.  I personally like reviewing the rendered HTML for large doc changes.

This PR is just a proof of concept (and a failing one at the moment).  My vision was that doc-build workflow would add one, and only one comment to the PR.  Force pushes to the PR would not generate a new comment.

I see other potential use cases for workflows adding a comment if the security concerns can be addressed.  For instance, adding a comment with a test coverage report for the PR.  We do this in a downstream repo (on gerrit), and it has been very effective at encouraging developers to write more tests.

**Pull request closed without merging**